src/libostree/ostree-bootconfig-parser.c

```diff
@@ -56,6 +59,87 @@ ostree_bootconfig_parser_clone (OstreeBootconfigParser *self)
   return parser;
 }
 
+/**
+ * ostree_bootconfig_parser_parse_tries:
+ * @self: Parser
+ * @path: File path
+ *
+ * Parses a suffix of two counters in the form "+LEFT-DONE" from the end of the
+ * filename (excluding file extension).
+ */
+static void
+parse_bootloader_tries (const char *filename, guint64 *out_left, guint64 *out_done)
+{
+  *out_left = 0;
+  *out_done = 0;
+
+  const char *counter = strrchr (filename, '+');
+  if (!counter)
+    return;
+  counter += 1;
+
+  guint64 tries_left = 0;
+  guint64 tries_done = 0;
+
+  {
+    char *endp = NULL;
+    tries_left = g_ascii_strtoull (counter, &endp, 10);
+    if (endp == counter || (tries_left == G_MAXUINT64 && errno == ERANGE))
+      return;
+    counter = endp;
+  }
+
+  /* Parse done counter only if present */
+  if (*counter == '-')
+    {
+      char *endp = NULL;
+      tries_done = g_ascii_strtoull (counter + 1, &endp, 10);
+      if (endp == counter || (tries_done == G_MAXUINT64 && errno == ERANGE))
+        return;
+    }
+
+  *out_left = tries_left;
+  *out_done = tries_done;
+}
+
+/**
+ * ostree_bootconfig_parser_get_tries_left:
+ * @self: Parser
+ *
+ * Returns: Amount of boot tries left
+ *
+ * Since: 2025.2
+ */
+guint64
+ostree_bootconfig_parser_get_tries_left (OstreeBootconfigParser *self)
+{
+  return self->tries_left;
+}
+
+/**
+ * ostree_bootconfig_parser_get_tries_done:
+ * @self: Parser
+ *
+ * Returns: Amount of boot tries
+ */
+guint64
+ostree_bootconfig_parser_get_tries_done (OstreeBootconfigParser *self)
+{
+  return self->tries_done;
+}
+
+/**
+ * ostree_bootconfig_parser_is_parsed:
+ * @self: Parser
+ *
+ * Returns: TRUE if the boot config was parsed from existing boot config file
+ */
+gboolean
+ostree_bootconfig_parser_is_parsed (OstreeBootconfigParser *self)
```

Comment:
Hmm how about storing the original filename instead? i.e. `ostree_bootconfig_parser_get_filename`. It seems more generally useful and easy to describe than what "is_parsed" means.